src/walletSdk/Horizon/transaction/TransactionBuilder.ts

```diff
@@ -82,4 +114,27 @@ export class TransactionBuilder {
     });
     return this.builder.build();
   }
+
+  transferWithdrawalTransaction(
+    transaction: WithdrawTransaction,
+    assetId: StellarAssetId,
```

Comment:
@acharb could we infer the asset from the transaction? I think having an asset should probably be mandatory for a transaction in the `pending_user_transfer_start` status. What do you think?

Reply:
hmm that's a good question. I'm leaning towards leaving it because according to [sep docs](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0024.md#shared-fields-for-both-deposits-and-withdrawals) the amount_in_asset field is optional, so to be safe

the kotlin sdk also has the `assetId` param like this, @Ifropc wdyt?

Reply:
> could we infer the asset from the transaction? I think having an asset should probably be mandatory for a transaction in the pending_user_transfer_start status

MGI and some other anchors... 
I'm not opposed to have 2 methods tho, one is simplified with just transaction, and the other one is this one

Reply:
I'm going to go in favor of simplicity. Imo a second method like `transferWithdrawalTransactionInfer` (since you can't have two methods w/ same name in ts) would be more confusing for the user

if either feel strongly though lmk